pkg/api/v20170701/validate.go

```diff
@@ -114,10 +114,11 @@ func handleValidationErrors(e validator.ValidationErrors) error {
 	err := e[0]
 	ns := err.Namespace()
 	switch ns {
-	case "Properties.OrchestratorProfile", "Properties.MasterProfile",
+	case "Properties.OrchestratorProfile", "Properties.OrchestratorProfile.OrchestratorType",
+		"Properties.MasterProfile",
 		"Properties.MasterProfile.DNSPrefix", "Properties.MasterProfile.VMSize",
 		"Properties.LinuxProfile", "Properties.ServicePrincipalProfile.ClientID",
-		"Properties.ServicePrincipalProfile.Secret", "Properties.WindowsProfile.AdminUsername",
+		"Properties.WindowsProfile.AdminUsername",
```

Comment:
Oh, we have Secret or KeyvaultSecretRef. We only need either of them to be there.

And also for vlab specific, looks we allow a ServicePrincipalProfile with all the fields empty for non-kubernetes scenario. I don't want to break the existing behavior. Although i don't think it is right. But this is more of a correction PR for that purpose. This PR is scoped to keep the existing behavior, only change how validate handled.